WC_Layers/OpenMC_R2S.py

```diff
@@ -0,0 +1,407 @@
+                if energy == None:
+                    photon_source = openmc.IndependentSource(
+                        energy = energy,
+                        particle = 'photon',
+                        strength = 0.0)
+                else:    
+                    photon_source = openmc.IndependentSource(
+                        energy = energy,
+                        particle = 'photon',
+                        strength = energy.integral())
+                
+            else:
+                photon_source = openmc.IndependentSource(
+                    energy = None,
+                    particle = 'photon',
+                    strength = 0.0)
+            
+            photon_sources[mat_index] = photon_source
+        all_photon_sources[int_index] = openmc.MeshSource(unstructured_mesh, photon_sources)
+    return all_photon_sources
+    
+def make_photon_tallies(coeff_geom, photon_model, num_cooling_steps):
+    dose_energy, dose = openmc.data.dose_coefficients('photon', geometry=coeff_geom) #[pSv cm^2]
+    dose_filter = openmc.EnergyFunctionFilter(dose_energy, dose)
+
+    spherical_mesh = openmc.SphericalMesh(np.arange(0, 1505, 5), origin = (0.0, 0.0, 0.0), mesh_id=2, name="spherical_mesh")
+    spherical_mesh_filter = openmc.MeshFilter(spherical_mesh)
+    
+    particle_filter = openmc.ParticleFilter('photon')
+    energy_filter_flux = openmc.EnergyFilter.from_group_structure("VITAMIN-J-42")
+
+    flux_tally = openmc.Tally(tally_id=1, name="photon flux tally")
+    flux_tally.filters = [spherical_mesh_filter, energy_filter_flux, particle_filter]
+    flux_tally.scores = ['flux']
+
+    dose_tally = openmc.Tally(tally_id=2, name="dose tally") #[particle-cm/src]
+    dose_tally.filters = [spherical_mesh_filter, energy_filter_flux, particle_filter, dose_filter]
+    dose_tally.scores = ['flux']
+
+    photon_model.tallies = [flux_tally, dose_tally]
+    #Change boundary condition to obtain flux/dose beyond original geometry:
+    list(photon_model.geometry.get_all_surfaces().values())[-1].boundary_type="transmission"
+
+    tally_sphere = openmc.Sphere(r = 3900)
+    tally_sphere.boundary_type = "reflective"
+    tally_cell = openmc.Cell(fill = None, region = +list(photon_model.geometry.get_all_surfaces().values())[-1] & -tally_sphere)
+    photon_model.geometry.root_universe.add_cell(tally_cell)
+
+    for int_index in range(num_cooling_steps):
+        #Reassign settings (which contains source) for each decay step
+        photon_model.settings = openmc.Settings.from_xml(f'settings_{int_index}.xml')
+        for source in photon_model.settings.source:
+            if "domain_ids" in source.constraints:
+                source.constraints['domain_ids'].append(source.constraints['domain_ids'][-1]+1)
+        photon_model.export_to_model_xml(path=f'photon_model_{int_index}.xml')
+        sp_path = photon_model.run(f'photon_model_{int_index}.xml')
+        sp_path.rename(f'statepoint_openmc_{int_index}.h5')
```

Comment:
Separation of concerns: these lines here are only tangentially related to the methods name.  They probably deserve to be in their own function(s). Even more, writing output should generally be separated from processing data.

This function may not need to take the `photon_model` as an input, and then just return the tally data, at which point it can be added to the model.

Then a separate function can expand the boundary of the model, and an other function can write the various cooling time models to XML.